### ISSUE TITLE
Hausdorff distance loss support

### ISSUE THREAD

**Issue by imadtoubal** [Project-MONAI/MONAI#6993]
Hausdorff distance is widely used in evaluating medical image segmentation methods. Adding an objective/loss function directly to optimize this distance can be instrumental in optimizing this score [1].

<center><img src="https://upload.wikimedia.org/wikipedia/commons/2/21/Hausdorff_distance_sample.svg" alt="Hausdorff Distance"></center>

An existing implementation is [publicly available on GitHub](https://github.com/PatRyg99/HausdorffLoss) by [Patryk Rygiel](https://github.com/PatRyg99), although it might take some engineering work to integrate within the MONAI framework.

I was able to train a 3D segmentation model successfully using MONAI's SwinUNETR with this implementation and I would like to contribute the code for this loss so that it's more widely available and easier to integrate with MONAI-based repositories.

References:
[1] Karimi, D., & Salcudean, S. E. (2019). [Reducing the Hausdorff distance in medical image segmentation with convolutional neural networks](https://ieeexplore.ieee.org/abstract/document/8767031?casa_token=OGugo3jEuXQAAAAA:y5fp0CuZiEWlbx8FS4XKHu43Wi_W7albLeIAjsnZ1c-IlTpJmEFJUgoMlKl_taSYuBpcPyCvtTMe). IEEE Transactions on medical imaging, 39(2), 499-513.

**Comment by imadtoubal** [comment]
Looks like I have forgotten to include the parameter alpha [in the constructor method docstring](https://github.com/Project-MONAI/MONAI/blob/dev/monai/losses/hausdorff_loss.py#L57). Do you recommend using this issue to make a PR to fix that?
